client/daemon/peer/peertask_conductor.go

```diff
@@ -910,7 +910,18 @@ func (pt *peerTaskConductor) pullSinglePiece() {
 	pt.SetTotalPieces(1)
 	pt.SetPieceMd5Sign(digest.SHA256FromStrings(pt.singlePiece.PieceInfo.PieceMd5))
 
-	request := &DownloadPieceRequest{
+	var (
+		err     error
+		request *DownloadPieceRequest
+		result  *DownloadPieceResult
+	)
+
+	err = pt.UpdateStorage()
+	if err != nil {
+		goto fallback
```

Comment:
is this necessary to use goto here? looks like can define a function to handle fallback?